### PR TITLE
Add GCP support

### PR DESCRIPTION
**Pull request by mjudeikis**
Add GCP support and refactor interface 

In the PR:
* revendor hive api
* revendor operator SDK
* vendor GCP DNS API
* move clients to more abstract subpackage
* make certman to be orientated around the new interface for LetsEncrypt

### CODE REVIEW

File: docs/developing.md [hunk 1]
```diff
@@ -44,11 +44,19 @@ oc create configmap certman-operator \
 Two Secrets are required. One is the AWS credentials that we'll need for working with Route53.
 
 ```bash
-oc create secret generic aws-iam-secret --from-literal=aws_access_key_id=XXX --from-literal=aws_secret_access_key=YYYY
+    # AWS
+    oc create secret generic aws-iam-secret --from-literal=aws_access_key_id=XXX --from-literal=aws_secret_access_key=YYYY
+    # GCP
+    oc create secret generic gcp-secret --from-file=$HOME/.gcp/osServiceAccount.jso
```
Comment (dofinn) [inline docs/developing.md]: .json

**Comment by mjudeikis** [comment]
@dofinn any luck with the review?

**Comment by c-e-brumm** [comment]
Hi @mjudeikis, we will need to clear some work to make time to test this PR. If this is a good time we can make an effort to start this week.

**Comment by mjudeikis** [comment]
@c-e-brumm let me know if I can help somehow

**Comment by dak1n1** [comment]
I tested this branch in minikube and it appears to be working. Certs were generated and revoked, as expected.

File: cmd/manager/main.go [hunk 2]
```diff
@@ -149,8 +155,11 @@ func main() {
...
-	if err := mgr.Start(signals.SetupSignalHandler()); err != nil {
-		log.Error(err, "Manager exited non-zero")
-		os.Exit(1)
+	return mgr.Start(stopCh)
+}
+
+func main() {
+	if err := start(); err != nil {
+		panic(err)
 	}
```
Comment (dak1n1) [inline cmd/manager/main.go]: @mjudeikis this section might be causing some problems in the staging environment. Here's the output from the pod:

```
[root@hive-stage-master-ip-10-140-2-38.ec2.internal ~]# oc get pods -n certman-operator
NAME                                READY     STATUS    RESTARTS   AGE
certman-operator-7d755788c8-5g7ns   0/1       Error     2          31s
certman-operator-catalog-5ln5b      1/1       Running   0          13m

[root@hive-stage-master-ip-10-140-2-38.ec2.internal ~]# oc logs certman-operator-7d755788c8-5g7ns  -n certman-operator
{"level":"info","ts":1575422149.6017106,"logger":"cmd","msg":"Go Version: go1.11.13"}
{"level":"info","ts":1575422149.6018276,"logger":"cmd","msg":"Go OS/Arch: linux/amd64"}
{"level":"info","ts":1575422149.6018362,"logger":"cmd","msg":"Version of operator-sdk: v0.8.2"}
{"level":"info","ts":1575422149.6021032,"logger":"leader","msg":"Trying to become the leader."}
{"level":"error","ts":1575422149.6386,"logger":"cmd","msg":"","error":"Unauthorized","stacktrace":"github.com/openshift/certman-operator/vendor/github.com/go-logr/zapr.(*zapLogger).Error\n\tsrc/github.com/openshift/certman-operator/vendor/github.com/go-logr/zapr/zapr.go:128\nmain.start\n\tsrc/github.com/openshift/certman-operator/cmd/manager/main.go:87\nmain.main\n\tsrc/github.com/openshift/certman-operator/cmd/manager/main.go:162\nruntime.main\n\t/usr/local/go/src/runtime/proc.go:201"}
panic: Unauthorized
goroutine 1 [running]:
main.main()
        src/github.com/openshift/certman-operator/cmd/manager/main.go:163 +0x44
```

Could you help with debugging this?

Comment (c-e-brumm) [reply cmd/manager/main.go]: FYI: Merge was reverted to bring staging back online.

Comment (mjudeikis) [reply cmd/manager/main.go]: Can I get access to staging to deploy manual build somehow? It a challenge to debug it it like this.

Comment (mjudeikis) [reply cmd/manager/main.go]: Error looks like leader election failed. Did you had old version still running?

Comment (dak1n1) [reply cmd/manager/main.go]: I tried deploying your changes in another stage cluster, and it works. So there probably isn't an issue with your code after all. There must be an issue with how it's deployed to hive-staging using OLM. I'm following up with my team about debugging OLM.

**Pull request merged**